packages/dev/core/src/Debug/physicsViewer.ts

```diff
@@ -183,6 +198,86 @@ export class PhysicsViewer {
         }
     }
 
+    protected _updateDebugConstraints() {
+        for (let i = 0; i < this._numConstraints; i++) {
+            const constraint = this._constraints[i];
+            const mesh = this._constraintMeshes[i];
+            if (constraint && mesh) {
+                this._updateDebugConstraint(constraint, mesh);
+            }
+        }
+    }
+
+    /**
+     * Given a scaling vector, make all of its components
+     * 1, preserving the sign
+     * @param scaling
+     */
+    protected _makeScalingUnitInPlace(scaling: Vector3) {
+        if (Math.abs(scaling.x - 1) > Epsilon) {
+            scaling.x = 1 * Math.sign(scaling.x);
+        }
+        if (Math.abs(scaling.y - 1) > Epsilon) {
+            scaling.y = 1 * Math.sign(scaling.y);
+        }
+        if (Math.abs(scaling.z - 1) > Epsilon) {
+            scaling.z = 1 * Math.sign(scaling.z);
+        }
+    }
+
+    protected _updateDebugConstraint(constraint: PhysicsConstraint, parentingMesh: AbstractMesh) {
+        if (!constraint._initOptions) {
+            return;
+        }
+
+        // Get constraint pivot and axes
+        const { pivotA, pivotB, axisA, axisB, perpAxisA, perpAxisB } = constraint._initOptions;
+
+        if (!pivotA || !pivotB || !axisA || !axisB || !perpAxisA || !perpAxisB) {
+            return;
+        }
+
+        parentingMesh.getDescendants(true).forEach((parentConstraintMesh) => {
+            // Get the parent transform
+            const parentCoordSystemNode = parentConstraintMesh.getDescendants(true)[0] as TransformNode;
+            const childCoordSystemNode = parentConstraintMesh.getDescendants(true)[1] as TransformNode;
+
+            const { parentBody, parentBodyIndex } = parentCoordSystemNode.metadata;
+            const { childBody, childBodyIndex } = childCoordSystemNode.metadata;
+
+            const parentTransform = this._getTransformFromBodyToRef(parentBody, TmpVectors.Matrix[0], parentBodyIndex);
+            const childTransform = this._getTransformFromBodyToRef(childBody, TmpVectors.Matrix[1], childBodyIndex);
+
+            parentTransform.decomposeToTransformNode(parentCoordSystemNode);
+            this._makeScalingUnitInPlace(parentCoordSystemNode.scaling);
+
+            childTransform.decomposeToTransformNode(childCoordSystemNode);
+            this._makeScalingUnitInPlace(childCoordSystemNode.scaling);
+
+            // Get the transform to align the XYZ axes to the constraint axes
+            const rotTransformParent = Quaternion.FromRotationMatrixToRef(

```

Comment:
Sorry, should have said that before :-)

Why not do it directly to the `parentTransformNode.rotationQuaternion` variable?

Reply:
(of course, same for the child)

Reply:
Makes sense! 🤣